go-controller/pkg/ovn/pods_retry.go

```diff
@@ -0,0 +1,170 @@
+package ovn
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// iterateRetryPods checks if any outstanding pods have been waiting for 60 seconds of last known failure
+// then tries to re-add them if so
+// updateAll forces all pods to be attempted to be retried regardless of the 1 minute delay
+func (oc *Controller) iterateRetryPods(updateAll bool) {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	now := time.Now()
+	for key, podEntry := range oc.retryPods {
+		if podEntry.ignore && podEntry.logicalPortCacheEntry == nil {
```

Comment:
this doesn't make sense to me. If the podEntry.ignore flag is set, it should ignore both add and deletion right?

Reply:
is this because you are trying to handle the case where there is a stale LSP that kapi deleted, but a new pod is being created with the same name?

Reply:
correct yes. I am not sure how else we can address that case of new pod with same name being created. So I purposefully used "ignore" for adds and "logicalPortCacheEntry" for deletes... If you noticed down in the deleteHandler, we don't call "unSkipRetry", we simply set the logicalPortCacheEntry that says it needs a delete. If we use the ignore flag for both adds and dels, it becomes hard for the new pod with same name case. Like there is no way for me to differentiate a "retry delete only" from a "retry delete and add" if we always set the ignore=false flag.

Reply:
spoke offline, we decided to add "needsAdd" flag.